src/rules.js

```diff
@@ -94,9 +94,23 @@ block.normal = merge({}, block);
 
 block.gfm = merge({}, block.normal, {
   nptable: /^ *([^|\n ].*\|.*)\n *([-:]+ *\|[-| :]*)(?:\n((?:.*[^>\n ].*(?:\n|$))*)\n*|$)/,
-  table: /^ *\|(.+)\n *\|?( *[-:]+[-| :]*)(?:\n((?: *[^>\n ].*(?:\n|$))*)\n*|$)/
+  table: '^ *\\|(.+)\\n' // Header
+    + ' *\\|?( *[-:]+[-| :]*)' // Align
+    + '(?:\\n((?:(?!^|>|\\n| |hr|heading|lheading|code|fences|list|html).*(?:\\n|$))*)\\n*|$)' // Cells
 });
 
+block.gfm.table = edit(block.gfm.table)
+  .replace('hr', block.hr)
+  .replace('heading', ' {0,3}#{1,6} +')
+  .replace('lheading', '([^\\n]+)\\n {0,3}(=+|-+) *(?:\\n+|$)')
+  .replace('blockquote', ' {0,3}>')
+  .replace('code', ' {4}[^\\n]+')
+  .replace('fences', ' {0,3}(?:`{3,}|~{3,})[^`\\n]*\\n')
```

Comment:
```suggestion
  .replace('fences', ' {0,3}(?:`{3,}[^`\\n]*|~{3,}[^\\n]*)\\n')
```
Looking at the other snippets I copied from the paragraph regex, I think the check for fences might also be out of spec. Looks like only code fences made with ` [cannot have backticks in the rest of the line](https://github.github.com/gfm/#example-115), but fences made with ~ [can have both backticks and tildes](https://github.github.com/gfm/#example-116).

Not sure if there's a cleaner way to make this fix, but If it's going into this PR it should also be fixed in the paragraph regex. Any thoughts?

Reply:
maybe this should be a separate pr looks like the fences regex should be changed to pass example 116:

```diff
- (`{3,}|~{3,})([^`~\n]*)\n
+ (`{3,}(?=[^`\n]*\n)|~{3,})([^\n]*)\n
```

Reply:
Does the lookahead `(?=` make the fence regex less prone to REDOS as well? I can't remember but it seems like lookaheads acted like atomic groups which probably helps.

Reply:
spec tests marked `shouldFail: true` (like example 116) are ones that are currently failing.

if you remove that line from the json file it will fail in the tests. With this change it will pass that example. I'm not sure about being less prone to redos.

Reply:
Here ya go: Lucky number #1600.

Reply:
👍 you should commit this change and remove the (`+`) on `heading` and this should be good to go once we hear back from @davisjam

Reply:
Committed.